cmd/snap-confine/udev-support.c

```diff
@@ -305,15 +307,34 @@ void sc_setup_device_cgroup(const char *security_tag)
 	struct udev_list_entry *assigned;
 	assigned = udev_enumerate_get_list_entry(devices);
 	if (assigned == NULL) {
-		/* NOTE: Nothing is assigned, don't create or use the device cgroup. */
-		debug("no devices tagged with %s, skipping device cgroup setup",
-		      udev_tag);
-		return;
+		if (mode == SC_DEVICE_CGROUP_MODE_OPTIONAL) {
+			/* NOTE: Nothing is assigned, don't create or use the device cgroup. */
+			debug
+			    ("no devices tagged with %s, skipping device cgroup setup",
+			     udev_tag);
+			return;
+		} else {
+			/* the device cgroup was requested to be set up despite of no
+			 * devices being assigned to this snap */
+			debug
+			    ("no devices tagged with %s, but device cgroup is required, proceeding with setup",
+			     udev_tag);
+		}
 	}
 
 	/* cgroup wrapper is lazily initialized when devices are actually
 	 * assigned */
 	sc_device_cgroup *cgroup SC_CLEANUP(sc_device_cgroup_cleanup) = NULL;
+
+	if (cgroup == NULL && mode == SC_DEVICE_CGROUP_MODE_REQUIRED) {
```

Comment:
don't we know already that cgroup is NULL here?

Reply:
good point, thanks